shell/server/proxy.js

```diff
@@ -36,6 +36,18 @@ SANDSTORM_ALTHOME = Meteor.settings && Meteor.settings.home;
 SANDSTORM_LOGDIR = (SANDSTORM_ALTHOME || "") + "/var/log";
 SANDSTORM_VARDIR = (SANDSTORM_ALTHOME || "") + "/var/sandstorm";
 
+// User-agent strings that should be allowed to use http basic authentication.
+// These are regex matches, so ensure they are escaped properly with double
+// backslashes.
+BASIC_AUTH_USER_AGENTS = [
+  "git\\/",
+  "GitHub-Hookshot\\/",
+  "mirall\\/",
+  "Mozilla\\/5\.0 \\([^\\]]*\\) mirall\\/",
```

Comment:
I think the `\` in `5\.0` should be doubled as well, otherwise Javascript interprets "5\.0" the same as "5.0" which is a regex that will also match "500", etc.

I don't quite understand the middle part of the regex. You're matching a string of any characters except `]`? Why `]`?